module-2/backend/week_two.md

```diff
@@ -26,20 +38,79 @@ end
 
 Now how would you find the owner of the team with an id of 4?
 
+Team.find(4).owner
+
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+has_many_and_belongs_to_many for both
+
+teachers      teacher-students      students
+id              teacher_id            id
+name            student_id            name
+
+
 6. Define foreign key, primary key, and schema.
+
+The primary key is the main id the db entry has in the table. It is usually going to be assigned as the entry is made in the db, unless it is specifically overridden.
+Foreign keys are ids from another table which establish relationships between db entries.
+The schema is the blueprint for how db relationships are structured, as well as what information will be in a given table.
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+
+It's a belongs_to relationship. The specific db entry belongs_to the foreign key entry.
+
 8. What are the parts of an HTTP response?
 
+Status, header, body.
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+
+::pluck grabs the specified thing(s). Like Team.all.pluck(:owner_id) will give an array of owner_ids for all the Team instances.
+
+::find finds by id by default, but can take options and find by any attribute. Finds the first one matching the criteria.
+
+::where is like find but returns all instances that match the criteria.
+
+::create calls `new` and `save` together.
+
+::group returns a hash with the object as the key and whatever arguments were given to group as the value.
+
+I kind of feel like ::group in particular doesn't make a lot of sense in a vacuum, it is super useful as part of a chain of calls. 
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+
+rake db:reset. It drops the tables, recreates them, and seeds them.
+rake db:rollback. It rolls back the last migration, but can take STEP=# options if you want to roll back more than one migration.
+rake db:seed. It's just nice to have some data to work with and not have to manually create it.
+
 3. What two columns does `t.timestamps null: false` create in our database?
+
+created at and updated at.
+
 4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+
+Schools have_many Teachers, Teachers belong_to Schools.
+
 5. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+
+Teachers                Schools
+id                        id
+school_id   
+
 6. Give an example of when you might want to store information besides ids on a join table.
+
+I'm not really sure.
```

Comment:
invoice_items is a good example; we store quantities and unit prices on it.

`enrollments` might serve as a join between students and courses. We could store a grade on that join.